metadb/config.go

```diff
@@ -0,0 +1,34 @@
+package metadb
+
+const (
+	defaultHTTPAddress = "0.0.0.0:4001"
+	defaultRaftAddress = "0.0.0.0:4002"
+	defaultDataDir     = ".data"
+)
+
+// Config contains settings of the rqlite server
+type Config struct {
+	// http server bind address. For HTTPS, set X.509 cert and key
+	HTTPAddress string `mapstructure:"http_address" json:"http_address,omitempty"`
+	// unique name for node. If not set, set to hostname
+	NodeID string `mapstructure:"node_id" json:"node_id,omitempty"`
+	// raft communication bind address
+	RaftAddress string `mapstructure:"raft_address" json:"raft_address,omitempty"`
+	// comma-delimited list of nodes, through which a cluster can be joined (proto://host:port)"
+	JoinAddress string `mapstructure:"join_address" json:"join_address,omitempty"`
+	// set Discovery Service URL
+	DiscoveryURL string `mapstructure:"discovery_url" json:"discovery_url,omitempty"`
+	// set Discovery ID. If not set, Discovery Service not used
+	DiscoveryID string `mapstructure:"discovery_id" json:"discovery_id,omitempty"`
+	// data directory for rqlite
+	DataDir string `mapstructure:"data_dir" json:"data_dir,omitempty"`
+}
```

Comment:
```suggestion
type Config struct {
         // IPv4 or IPv6 address for http server and rafd
	ListenAddress string `mapstructure:"listen_address" json:"listen_address,omitempty"`
	
	// http server bind port. For HTTPS, set X.509 cert and key
	HTTPPort string `mapstructure:"http_port" json:"http_port,omitempty"`
	
	// raft port communication bind address
	RaftPort string `mapstructure:"raft_port" json:"raft_port,omitempty"`	
		
	// unique name for node. If not set, set to hostname
	NodeID string `mapstructure:"node_id" json:"node_id,omitempty"`
	
	// set Discovery Service URL
	DiscoveryURL string `mapstructure:"discovery_url" json:"discovery_url,omitempty"`
	
	// set Discovery ID. If not set, Discovery Service not used
	DiscoveryID string `mapstructure:"discovery_id" json:"discovery_id,omitempty"`
	
	// data directory for rqlite
	DataDir string `mapstructure:"data_dir" json:"data_dir,omitempty"`
}
```

1. Supernode is assumed to work in the same IP address, it makes no sense to give the opportunity to specify different HTTP and Raft addresses. Moreover I think we should to reduce settings to give the opportunity to specify only one listen_address for all services. I will do it later. but you please leave only one `ListenAddress`. Don't forget to update configs.

2. Not need to specify `JoinAddress`. There is no centralized server, you need to get the list of supernodes from Pastel RPC API, and try to connect automatically. For now please remove it. But create a task in Freedcamp, `Implement connection to rqlite cluster`

Reply:
task is created

Reply:
done